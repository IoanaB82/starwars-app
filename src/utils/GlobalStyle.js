import { createGlobalStyle } from "styled-components";

const GlobalStyle = createGlobalStyle`
    * {
        margin: 0;
        padding: 0;
    }

    :root {
        --color-dark-brown: #3d3336;
        --color-dark-blue: #4c596f;
        --color-light-brown: #ab856e;
        --color-light-blue-1: #6a8eb9;
        --color-light-blue-2: #bcc8dd;
      
        @import url(//db.onlinewebfonts.com/c/c00548f5d217ed324ae9096051927001?family=STARWARS);
      }

      body {
        
        box-sizing: content-box;
        margin: 0 auto;
        text-align: center;
        background-color: black;
        font-family: "STARWARS";
        height: auto;
      
        font-size: 18px;
      }

      
`;

export default GlobalStyle;
