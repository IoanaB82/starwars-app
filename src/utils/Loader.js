import React from "react";

const Loader = () => {
  return (
    <div
      style={{
        backgroundColor: "#1b1b1b",
        position: "absolute",
        top: "50%",
        left: "50%",
        transform: "translate(-50%)",
      }}
    >
      <img src="/loader.gif" alt="loader" />
    </div>
  );
};

export default Loader;
