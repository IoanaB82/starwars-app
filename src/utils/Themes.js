export const themes = {
  empire: {
    background: "#3D3336",
    headerBackground: "#698FBA",
    cardHeader: "#F1F3F2",
    cardBackground: "#AB856E",
  },

  revenge: {
    background: "#484445",
    headerBackground: "#B64419",
    cardHeader: "#595959",
    cardBackground: "#B9A79B",
  },

  attack: {
    background: "#373738",
    headerBackground: "#72979F",
    cardHeader: "#6E513D",
    cardBackground: "#B8805C",
  },
};
