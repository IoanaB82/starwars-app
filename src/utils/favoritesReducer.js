const reducer = (state, action) => {
  switch (action.type) {
    case "ADD_FAV":
      return state.includes(action.name) ? state : [...state, action.name];

    case "DELETE_FAV":
      return state.filter((item) => item !== action.name);

    default: {
      return state;
    }
  }
};

export default reducer;
