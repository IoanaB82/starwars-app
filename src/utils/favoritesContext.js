import React, { createContext, useReducer } from "react";

import favoritesReducer from "./favoritesReducer";

let key = "favs";

const initialFavsFunction = () => {
  if (typeof window !== "undefined") {
    const saved = window.sessionStorage.getItem(key);
    if (saved !== null) {
      return JSON.parse(saved);
    }
  }
  return [];
};

export const FavoritesContext = createContext();

export function FavoritesProvider(props) {
  let initialFavs = initialFavsFunction();
  const [favs, dispatch] = useReducer(favoritesReducer, initialFavs);

  return (
    <FavoritesContext.Provider value={{ favs, dispatch }}>
      {props.children}
    </FavoritesContext.Provider>
  );
}
