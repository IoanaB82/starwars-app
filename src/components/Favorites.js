import React, { useContext } from "react";
import styled from "styled-components";

import { FavoritesContext } from "../utils/favoritesContext";
import PageTitle from "./PageTitle";

const FavsCard = styled.div`
  display: flex;
  flex-direction: column;
  &:not(:last-child) {
    margin-right: 20px;
  }
`;

const Favorites = (props) => {
  const { favs, dispatch } = useContext(FavoritesContext);

  //console.log("in favorites component" + favsContext.favs);

  return (
    <>
      <PageTitle pageTitle={"Your Favorites"} />

      <div
        style={{
          display: "flex",
          flexWrap: "wrap",
        }}
        data-testid="favContent"
      >
        {favs.length > 0 ? (
          favs.map((el) => (
            <FavsCard style={{ padding: "20px" }}>
              <h3
                style={{
                  color: "white",
                  fontWeight: "lighter",
                  marginBottom: "20px",
                }}
              >
                {el}
              </h3>
              <img
                style={{ width: "300px", display: "inline-block" }}
                src={`/img/${el.replace(/[\s']/g, "").toLowerCase()}.jpg`}
                alt={`${el}`}
              />
              <button
                style={{
                  backgroundColor: "white",
                  padding: "10px",
                  fontFamily: "courier",
                  fontSize: "16px",
                  fontWeight: "600",
                  color: "red",
                  border: "none",
                }}
                onClick={() => dispatch({ type: "DELETE_FAV", name: el })}
              >
                Remove
              </button>
            </FavsCard>
          ))
        ) : (
          <h3> You have no favorites yet.</h3>
        )}
      </div>
    </>
  );
};

export default Favorites;
