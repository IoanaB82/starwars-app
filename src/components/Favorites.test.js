import React from "react";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import { MemoryRouter } from "react-router-dom";

import Favorites from "./Favorites";
import { FavoritesProvider } from "../utils/favoritesContext";

afterEach(cleanup);

function renderFavorites(args) {
  let defaultProps = {
    favs: [],
    location: "",
    errors: {},
  };

  const props = { ...defaultProps, ...args };

  return render(
    <MemoryRouter>
      <FavoritesProvider value={props.favs}>
        <Favorites {...props} />
      </FavoritesProvider>
    </MemoryRouter>
  );
}

//test that component renders
it("renders without crashing", () => {
  const { asFragment } = render(
    <MemoryRouter>
      <FavoritesProvider value={[]}>
        <Favorites />
      </FavoritesProvider>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});

//test that page title is shown correctly
it("should render a pagetitle component with content Favorites", () => {
  const { getByTestId } = renderFavorites();

  expect(getByTestId("pageTitle")).toHaveTextContent("Favorites");
});

//test what shows when favs array is empty, no favorites are chosen

it("expects to find You have no favorites when empty array", () => {
  const { getByTestId } = render(
    <MemoryRouter>
      <FavoritesProvider value={[]}>
        <Favorites />
      </FavoritesProvider>
    </MemoryRouter>
  );
  expect(getByTestId("favContent")).toHaveTextContent(
    "You have no favorites yet."
  );
});

//test that an element with name from favs array is shown

// it("expects to find a favorites card with title as array element", () => {
//   const favs = ["Hutt", "Droid"];
//   const { getByTestId } = render(
//     <MemoryRouter>
//       <FavoritesProvider value={favs}>
//         <Favorites />
//       </FavoritesProvider>
//     </MemoryRouter>
//   );
//   expect(getByTestId("favContent")).toHaveTextContent("Hutt");
// });
