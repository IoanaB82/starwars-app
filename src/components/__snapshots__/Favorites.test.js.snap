// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<DocumentFragment>
  <h1
    data-testid="pageTitle"
    style="color: white; padding: 30px; grid-row: 1;"
  >
    Your Favorites
  </h1>
  <div
    data-testid="favContent"
    style="display: flex; flex-wrap: wrap;"
  >
    <h3>
       You have no favorites yet.
    </h3>
  </div>
</DocumentFragment>
`;
